usaspending_api/references/models/agency.py

```diff
@@ -3,22 +3,19 @@
 
 
 class Agency(models.Model):
-
-    id = models.AutoField(primary_key=True)  # meaningless id
-    create_date = models.DateTimeField(auto_now_add=True, blank=True, null=True)
-    update_date = models.DateTimeField(auto_now=True, null=True)
-
-    toptier_agency = models.ForeignKey("references.ToptierAgency", models.DO_NOTHING, null=True, db_index=True)
-    subtier_agency = models.ForeignKey("references.SubtierAgency", models.DO_NOTHING, null=True, db_index=True)
-
-    # 1182 This flag is true if toptier agency name and subtier agency name are equal.
-    # This means the award is at the department level.
+    id = models.AutoField(primary_key=True)
+    create_date = models.DateTimeField(auto_now_add=True)
+    update_date = models.DateTimeField(auto_now=True)
+    toptier_agency = models.ForeignKey("references.ToptierAgency", models.DO_NOTHING, db_index=True)
```

Comment:
Why is toptier_agency using `models.ForeignKey` and subtier_agency is using `models.OneToOneField`?

Reply:
Because agency => toptier is many to one and subtier => agency is one to one.  Is this a trick question?

And to answer your followup question, the final design removes the agency table altogether, but we still have some data issues to sort out so it needs to stay this way for a bit longer.  Unfortunately, the currently-four-but-soon-to-be-three DoD agencies don't have subtiers so we otherwise wouldn't be able to represent them in the agency table without this structure.

And we currently need the agency table.  In the final design, the front end will need a good bit of rework because they're currently dependent on the agency surrogate id.  I want to get rid of that dependency eventually.

Hopefully somewhere in there is the answer to your question.

Reply:
No trick question, just still learning about agencies.